examples/src/main/java/com/teamdev/jxbrowser/examples/ScreenSharing.java

```diff
@@ -60,29 +67,72 @@ public static void main(String[] args) {
             tell.selectSource(screen, AudioCaptureMode.CAPTURE);
         });
 
-        // Subscribe to the capture session start event.
-        browserStreamer.on(CaptureSessionStarted.class, event ->
+        invokeLater(() -> {
+            JFrame frame = new JFrame("Screen Sharing Example");
+            JButton share = new JButton("Share My Screen...");
 
-            // Navigate to the screen sharing URL in the second browser.
-            browserSpectator.navigation().loadUrl(WEBRTC_SCREEN_SHARING_URL)
-        );
+            share.addActionListener(e -> {
 
-        initBrowserView(browserStreamer);
-        initBrowserView(browserSpectator);
+                // Load the WebRTC Screen Sharing Demo
+                // and wait until the document in the main frame is loaded completely.
+                loadUrlAndWaitForDocument(browser, WEBRTC_SCREEN_SHARING_URL);
 
-        browserStreamer.navigation().loadUrl(WEBRTC_SCREEN_SHARING_URL);
-    }
+                // Click the "Share Your Screen" button to start a capture session.
+                browser.mainFrame().ifPresent(mainFrame ->
+                        mainFrame.executeJavaScript(
+                                "document.getElementById('share-screen').click();"));
 
-    private static void initBrowserView(Browser browser) {
-        invokeLater(() -> {
-            BrowserView view = BrowserView.newInstance(browser);
+                showSuccessScreenSharingDialog(frame, browser.url());
+            });
 
-            JFrame frame = new JFrame("Screen Sharing Example");
             frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
-            frame.add(view, BorderLayout.CENTER);
-            frame.setSize(1200, 800);
+            frame.setSize(500, 300);
+            frame.setLayout(new GridBagLayout());
+            frame.add(share, new GridBagConstraints());
             frame.setLocationRelativeTo(null);
             frame.setVisible(true);
         });
     }
+
+    private static void loadUrlAndWaitForDocument(Browser browser, String url) {
+        CountDownLatch documentLoaded = new CountDownLatch(1);
+        Subscription subscription = browser.navigation().on(FrameDocumentLoadFinished.class, e -> {
```

Comment:
Why do you use `FrameDocumentLoadFinished` and not just `loadUrlAndWait()`?

Reply:
`loadUrlAndWait()` waits for the frame to finish loading, but not the page document. We need to make sure the document in the frame is completely loaded before interacting with it.

Reply:
We can interact with the document after `loadUrlAndWait()` and we don't need to wait for this event. Of it doesn't work in this case?

Reply:
In this case, document nodes have not loaded yet. So the javascript code just won't run.

Reply:
It's strange 🤔. Let me see what's going on.